go/adbc/README.md

```diff
@@ -0,0 +1,251 @@
+# adbc
+
+The adbc package provides:
+
+- Go Interfaces to [Arrow Database Connectivity](https://arrow.apache.org/adbc) (ADBC) for building ADBC drivers
+- An ADBC Driver Manager package (`drivermgr`) for loading and using ADBC Drivers
+- Scaffolding for writing new drivers in Go (`pkg`)
+
+See the [ADBC Website](https://arrow.apache.org/adbc/current/index.html) for more information on ADBC.
+
+## Getting Started
+
+To use ADBC with your database, you will need:
+
+1. An ADBC driver for your database
+2. The ADBC `drivermgr` package to load and use the driver
+
+Below we show how to use the `drivermgr` package to load and use the SQLite ADBC Driver.
+Not shown is how to make the SQLite ADBC driver available to the `drivermgr`.
+
+### Installation
+
+```sh
+go get github.com/apache/arrow-adbc/go/adbc/drivermgr
+```
+
+### Creating a Connection
+
+```go
+import (
+    "context"
+    "github.com/apache/arrow-adbc/go/adbc"
+    "github.com/apache/arrow-adbc/go/adbc/drivermgr"
+)
+
+ctx := context.Background()
+
+// Create driver and database
+var drv drivermgr.Driver
+db, err := drv.NewDatabase(map[string]string{
+    "driver": "adbc_driver_sqlite",
+})
+if err != nil {
+    return err
+}
+defer db.Close()
+
+// Open connection
+conn, err := db.Open(ctx)
+if err != nil {
+    return err
+}
+defer conn.Close()
```

Comment:
put this inside of a function of some kind for the example, such as `func main()` or otherwise. Just so it's valid and easily copy and pastable

Or separate the `import` section and the code into separate code blocks. It just looks awkward currently lol

Reply:
yeah, I'm not sure I love any one way of doing this. I'll try modelling it after how the literate readme for C++ is done and see if we like it (even if we can't test it).